src/apply-delta.ts

```diff
@@ -14,6 +14,16 @@ export type ApplyCurrentDate<O, U> = U extends { $currentDate: infer D }
     : O
   : O
 
+export type ApplyPush<O, U> = U extends { $push: infer D }
+  ? D extends Record<string, any>
+    ? O & ExpandPathObject<{
+      [K in keyof D]: D[K] extends { $each: infer E }
```

Comment:
Only $each can change the type of the final output, so it's the only special sub operator for push we need to worry about in this type.